### ISSUE TITLE
Export RESTINIO_USE_BOOST_ASIO define when using boost.

### ISSUE THREAD

**Issue by rcane** [Stiffstream/restinio-conan#4]
If using boost the conan package should export the `RESTINIO_USE_BOOST_ASIO` define so a consumer does not have to do it manually.

```python
def package_info(self):
  if self.options.boost_libs != "none":
    self.cpp_info.defines = ["RESTINIO_USE_BOOST_ASIO"]
```

**Comment by ngrodzitski** [comment]
Can you, please, show haw you set up your cmake-project and declare dependency for RESTinio?

Because, when you install restinio with `-o boost_libs=static` or `-o boost_libs=shared` there is the following properties defined for restinio target in file `<conan_storage_data>/restinio/0.4.9/stiffstream/stable/package/<HASH>/lib/cmake/restinio/restinio-targets.cmake`:
```cmake
set_target_properties(restinio::restinio PROPERTIES
  INTERFACE_COMPILE_DEFINITIONS "RESTINIO_USE_BOOST_ASIO"
  INTERFACE_COMPILE_FEATURES "cxx_std_14"
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
)
```
So this seems like propagating `RESTINIO_USE_BOOST_ASIO` definition to dependent targets.

**Comment by rcane** [comment]
I am new to conan so maybe I am doing something wrong.

My conanfile.txt looks like this:

```
[requires]
...
restinio/0.4.9@stiffstream/stable
...

[generators]
cmake

[options]
restinio:boost_libs=static
```

And with 
```bash
>conan install <source dir>
```
I create a cmake file which I include into my own.

```cmake
include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup(TARGETS)
```

Doing this with the current restinio recipe I get an error that `<asio.hpp>` cannot be found. And this led me to the discovery that `RESTINIO_USE_BOOST_ASIO` was not defined. Adding the `package_info` lines to the recipe fixes this for me.
The couple of conan recipes I had to write myself (which I copy-pasted from others) do it like this as well when defines are needed for the consumer of the package.

**Comment by eao197** [comment]
@rcane Have you seen [that example](https://github.com/Stiffstream/restinio-conan-example)?

**Comment by rcane** [comment]
I have seen this example. What I did not see was that you call `find_package(restinio CONFIG REQUIRED)` explicitly in your CMakeLists.txt. My understanding was that this was not necessary when using the conan-generated file. And in fact I don't use `find_package` at all for any of my external libraries. But this could explain where the `RESTINIO_USE_BOOST_ASIO` define comes from.

**Comment by rcane** [comment]
The actual including of RESTinio looks like this:

```cmake
target_link_libraries(foo
    ...
    CONAN_PKG::restinio
    ...
)
```

**Comment by prince-chrismc** [comment]
Issue could be closed.
Fixed by f72c3a4
